yaksh/views.py

```diff
@@ -605,7 +606,7 @@ def course_detail(request, course_id):
     user = request.user
     if not is_moderator(user):
         raise Http404('You are not allowed to view this page')
-    course = get_object_or_404(Course, creator=user, pk=course_id)
+    course = Course.objects.filter(Q(creator=user)|Q(teachers=user), pk=course_id)[0]
```

Comment:
function "course_detail" will be called only when there is a course either as a teacher or moderator. so list indexing will not throw error.